scripts/docker-secrets.sh

```
@@ -0,0 +1,54 @@
+# URL-safe tokens (chars [-_A-Za-z0-9]) are safe inside the postgres:// / redis:// URLs.
+gen() { python3 -c "import secrets; print(secrets.token_urlsafe($1))"; }
+# Matches the entrypoint's ENCRYPTION_MASTER_KEY format: base64(32 random bytes).
+enc() { python3 -c "import secrets,base64; print(base64.b64encode(secrets.token_bytes(32)).decode())"; }
+
+PG=$(gen 24); RD=$(gen 24); GF=$(gen 18); JWT=$(gen 64); JWTR=$(gen 64); ENC=$(enc)
+
+cp "$TEMPLATE" "$OUT"
+
+# Secrets go via stdin (newline-delimited), not argv — argv is world-readable in
+# /proc/<pid>/cmdline and `ps` while the process runs.
+printf '%s\n' "$PG" "$RD" "$GF" "$JWT" "$JWTR" "$ENC" | python3 - "$OUT" <<'PY'
```

Shellcheck (0.11.0)

[error] This redirection overrides piped input. To use both, merge or pass filenames.

(SC2259)